extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -101,16 +142,52 @@ public void initialize() {
         }
 
         DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
-        builder.setPskStore(new StaticPskStore("", configuration.code.getBytes()));
+        builder.setPskStore(new StaticPskStore(configuration.identity, configuration.preSharedKey.getBytes()));
         dtlsConnector = new DTLSConnector(builder.build());
         endPoint = new TradfriCoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
         deviceClient.setEndpoint(endPoint);
         updateStatus(ThingStatus.UNKNOWN);
 
         // schedule a new scan every minute
-        scanJob = scheduler.scheduleWithFixedDelay(() -> {
-            startScan();
-        }, 0, 1, TimeUnit.MINUTES);
+        scanJob = scheduler.scheduleWithFixedDelay(this::startScan, 0, 1, TimeUnit.MINUTES);
+    }
+
+    protected String authenticateOnGateway(TradfriGatewayConfig configuration) {
+        String preSharedPsk = null;
+        CoapResponse gatewayResponse;
+        try {
+            DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
+            builder.setPskStore(new StaticPskStore("Client_identity", configuration.code.getBytes()));
+
+            DTLSConnector dtlsConnector = new DTLSConnector(builder.build());
+            CoapEndpoint authEndpoint = new CoapEndpoint(dtlsConnector, NetworkConfig.getStandard());
+            String authUrl = "coaps://" + configuration.host + ":" + configuration.port + "/15011/9063";
+
+            CoapClient deviceClient = new CoapClient(new URI(authUrl));
+            deviceClient.setEndpoint(authEndpoint);
+
+            JsonObject json = new JsonObject();
+            json.addProperty(CLIENT_IDENTITY_PROPOSED, configuration.identity);
+
+            gatewayResponse = deviceClient.post(json.toString(), 0);
+
+            authEndpoint.destroy();
+            deviceClient.shutdown();
+
+            if (gatewayResponse.isSuccess()) {
+                String responseText = gatewayResponse.getResponseText();
+                json = new JsonParser().parse(responseText).getAsJsonObject();
+                preSharedPsk = json.get(NEW_PSK_BY_GW).getAsString();
+            } else {
+                logger.error("Can't obtain pre-shared key for identity => {} (response code => {}, response text => {})",
+                        configuration.identity, gatewayResponse.getCode(),
+                        isNullOrEmpty(gatewayResponse.getResponseText()) ? "<empty>" : gatewayResponse.getResponseText());
+            }
+        } catch (URISyntaxException | JsonParseException e) {
+            logger.error(String.format("Can't obtain pre-shared key for identity => %s", configuration.identity), e);
```

Comment:
I would consider URISyntaxException a bug in this code and thus an ERROR is ok.
JsonParseException is not a bug on our end, so there should not be any ERROR logging in that case.

Reply:
Again - what severity would you suggest. (From my perspective and error is a condition which prevents the normal workflow of the process - here - pairing the tradfri bridge - no matter if the issue is in the device itself, in the code or in the network connection between both. In either case - the pairing can't proceed so - it's an error)

Reply:
Please check [E.5 from our guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#e-logging) to better understand how to use the logging levels.